include/circt/Dialect/RTL/RTLTypes.td

```diff
@@ -86,3 +86,66 @@ class InOutElementConstraint<string value, string inoutValue>
   : TypesMatchWith<"type should be element of inout type",
                    inoutValue, value,
                    "$_self.cast<InOutType>().getElementType()">;
+
+// Adapted from TestTypeDefs.td
+// A parent type for any type which is just a list of fields (e.g. structs,
+// unions).
+def StructType : RTLType<"Struct"> {
+  let summary = "RTL struct type";
+  let description = [{
+    Struct type has value semantics, updates to the struct are logically by 
+    copy.
+  }];
+  let mnemonic = "struct";
+
+  let parameters = (
+    ins
+    // An ArrayRef of something which requires allocation in the storage
+    // constructor.
+    ArrayRefOfSelfAllocationParameter<
+      "::circt::rtl::StructType::FieldInfo",
+      "struct fields">: $elements
+  );
+
+  // Prints the type in this format:
+  //   struct<[{field1Name, field1Type}, {field2Name, field2Type}]>
```

Comment:
Did you consider a syntax that uses colons and fewer curlies?

`struct<field1: i32, field2: f64, field2: struct<a: i32, b: i32>>`

?